File: composition/java/Square.java
public class Square implements Rectangular {
  private Rectangle rect;

  public Square(int side, int x, int y) {
    rect = new Rectangle(side, side, x, y);
  }

  public int getWidth() {
    return rect.getWidth();
  }

  public Rectangular setWidth(int size) {
    return setSide(size);
  }

  public int getHeight() {
    return rect.getHeight();
  }

  public Rectangular setHeight(int size) {
    return setSide(size);
  }

  public int getSide() {
    return rect.getWidth();
  }

  public Rectangular setSide(int size) {
    rect.setWidth(size);
    rect.setHeight(size);
    return this;
  }

  public int getX() {
    return rect.getX();
  }

  public int getY() {
    return rect.getY();
  }

  public int perimeter() {
    return rect.perimeter();
  }

  public int area() {
    return rect.area();
  }

  public Rectangle getRectangle() {
    return rect;
  }

  public Rectangular augmentSideBy(int size) {
    rect.augmentWidthBy(size);
    rect.augmentHeightBy(size);
    return this;
  }

  public Rectangular augmentWidthBy(int size) {
    return augmentSideBy(size);
  }

  public Rectangular augmentHeightBy(int size) {
    return augmentSideBy(size);
  }

  public Rectangular rotate() {
    return rect.rotate();
  }

  public String toString() {
    return "Square:\n\twidth: " + getWidth() + "\theight: " + getHeight() +
    "\n\tx: " + getX() + "\t\ty: " + getY() +
    "\n\tperimeter: " + perimeter() + "\tarea: " + area();
  }
}

File: composition/ruby/square.rb
require 'forwardable'

class Square
  extend Forwardable

  def_delegators :@rect, :width, :height, :x, :y,
    :perimeter, :area, :rotate, :rotate!

  def initialize(side, x, y)
    @rect = Rectangle.new side, side, x, y
  end

  def side
    return @rect.width if @rect.width == @rect.height
    raise StandardError, "Something went wrong with this square(width of #{@rect.width} should equal height of #{@rect.height}"
  end

  def side=(size)
    rect.width, rect.height = size, size
  end

  def augment_side_by(size)
    Square.new side + size, top_left_corner
  end

  def augment_side_by!(size)
    @rect.augment_width_by! size
    @rect.augment_height_by! size
    self
  end

  def rectangle
    @rect.dup
  end

  def to_s
    <<-EOF
Square:
width: #{width}\theight: #{height}
x: #{x}\t\ty: #{y}
perimeter: #{perimeter}\tarea: #{area}
    EOF
  end
end
